src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -174,6 +181,117 @@ public void connect() {
+            new BufferedReader(
+                  new InputStreamReader(importProcess.getInputStream())
+            ).lines().filter(line -> line.startsWith(idLinePrefix)).map(line -> line.substring(idLinePrefix.length())).collect(Collectors.toList());
+
+        if (relevantOutput.isEmpty()) {
+          return null;
+        }
+
+        Set<Long> imageIds = Arrays.stream(relevantOutput.get(0).split(",")).map(Long::parseLong).collect(
+            Collectors.toSet());
+        return imageIds;
+
+      } else {
+        // nothing was imported so we also return no IDs
+        return null;
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    return null;
+  }
+
+  /**
+   * Searches for an omero project for the current user
+   * @param projectName the name of the project which should be searched
+   * @param datasetName the name of the dataset that is searched
+   * @return the omero ID for the dataset matching search criteria, null if the search found no match
+   */
+  private Long findOmeroDataset(String projectName, String datasetName) {
+    if (!this.isConnected()) {
+      this.connect();
+    }
+    try {
+      BrowseFacility browseFacility = gateway.getFacility(BrowseFacility.class);
+      Collection<ProjectData> projects = browseFacility.getProjects(securityContext);
+      List<ProjectData> matchingProjects = projects.stream().filter(projectData -> projectData.getName().equals(projectName)).collect(
+          Collectors.toList());
+
+      Set<DatasetData> projectDatasets = new HashSet<>();
+
+      if (matchingProjects.isEmpty()) {
+        return null;
+      } else if (matchingProjects.size() > 1) {
+        for (ProjectData projectData: matchingProjects) {
+            projectDatasets.addAll(projectData.getDatasets());
+        }
+      }
+
+      if (projectDatasets.isEmpty()) {
+        return null;
+      }
+
+      List<DatasetData> matchingDatasets = projectDatasets.stream().filter(datasetData -> datasetData.getName().equals(datasetName)).collect(Collectors.toList());
+
+      // dataset identifiers should be unique and we need to abort when we find more than one datast matching the criteria
+      if (matchingDatasets.isEmpty()) {
+        return null;
+      } else if (matchingDatasets.size() > 1) {
+        throw new IllegalArgumentException("There are multiple datasets with name " + datasetName + " for project " + projectName);
+      }
+    } catch (DSOutOfServiceException dsOutOfServiceException) {
+      throw new RuntimeException("Error while accessing omero service: broken connection, expired session or not logged in", dsOutOfServiceException);
+    } catch (ExecutionException executionException) {
+      throw new RuntimeException("Task aborted unexpectedly.", executionException);
+    } catch (DSAccessException dsAccessException) {
+      throw new RuntimeException("Could not pull data from the omero server.", dsAccessException);
+    }
+    return null;
+  }
```

Comment:
you should add a case that actually returns an ID if one dataset is found 😄

Reply:
yikes. My bad.

Reply:
a7229ac